BootloaderCommonPkg/Library/ContainerLib/ContainerLib.c

```diff
@@ -778,7 +782,14 @@ LoadComponentWithCallback (
   Status = AuthenticateComponent (CompBuf, SignedDataLen, AuthType,
              CompData + ALIGN_UP(SignedDataLen, AUTH_DATA_ALIGN),  HashData, Usage);
   if (LoadComponentCallback != NULL) {
-    LoadComponentCallback (PROGESS_ID_AUTHENTICATE);
+    // Update component Call back info afte authenticaton is done
+    // This info will used by firmware stage to extend to TPM
+    CbInfo.ComponentType    = ComponentId;
+    CbInfo.CompBuf          = CompBuf;
+    CbInfo.CompLen          = SignedDataLen;
+    CbInfo.HashAlg          = GetHashAlg(AuthType);
+    CbInfo.HashData         = HashData;

```

Comment:
How do you know the HashData is valid since the Status returned by Auth() could be error ?  
Should we consider only extending hash when AuthenticateComponent() returns  SUCCESS ?

Reply:
This callback would be invoked only for
- Stage2 authentication where hashdata would be NULL
- Payload and Epayload where hashdata is valid only for Epayload
Incase authentication fails, boot would halt. 
Added a additional check to invoke CB only Auth success cases